### PR TITLE
Adding further validation and logging

### PR DESCRIPTION
**Pull request by TayGov**
# Description

This PR includes the following proposed change(s):

Adding restricted document validation
Adding logging for failed document checks

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Unit test and postman

## Does the change impact or break the Docker build?

- [ ] Yes
- [X] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/error/AiReviewerRestrictedDocumentException.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package ca.bc.gov.open.jag.efilingreviewerapi.error;
+
+public class AiReviewerRestrictedDocumentException extends AiReviewerException {
```
[the remark this answers is not in the document]
Comment (TayGov) [reply src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/error/AiReviewerRestrictedDocumentException.java:3]: Following the project standard naming

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java [hunk 2]
```diff
@@ -15,6 +15,9 @@ private Keys() {}
     public static final Map<String, String> ACCEPTED_DOCUMENT_TYPES = new HashMap<String, String>() {{
         put("RCC", "Response to Civil Claim");
     }};
+    public static final Map<String, String> RESTRICTED_DOCUMENT_TYPES = new HashMap<String, String>() {{
```
[the remark this answers is not in the document]
Comment (TayGov) [reply src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java:18]: This is static

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java [hunk 3]
```diff
@@ -15,6 +15,9 @@ private Keys() {}
...
+        put("TEST", "This is a temporary");
```
Comment (akroon3r) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java:19]: This is a temporary test?

File: src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/document/validators/documentValidator/ValidateExtractedDocumentTest.java [hunk 4]
```diff
@@ -19,6 +20,7 @@
     public static final String DOCUMENT_TYPE = "RCC";
     public static final String RESPONSE_TO_CIVIL_CLAIM = "Response to Civil Claim";
     public static final String NOT_RESPONSE_TO_CIVIL_CLAIM = "THIS IS NOT VALID";
+    public static final String RESTRICTED_DOCUMENT = "This is a temporary";
```
Comment (akroon3r) [inline src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/document/validators/documentValidator/ValidateExtractedDocumentTest.java:23]: same

**Review by akroon3r — COMMENTED** [review]
one question

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java (hunk 3 above)
Comment (TayGov) [reply src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/Keys.java:19]: I don't have any of the restricted document types. So for now I put a value in.

File: src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/document/validators/documentValidator/ValidateExtractedDocumentTest.java (hunk 4 above)
Comment (TayGov) [reply src/backend/efiling-reviewer-api/src/test/java/ca/bc/gov/open/jag/efilingreviewerapi/document/validators/documentValidator/ValidateExtractedDocumentTest.java:23]: This is to test that it "hits" a restricted document

**Review by akroon3r — APPROVED** [review]

**Pull request merged**